framework/src/util/CFRLogger.cpp

```diff
@@ -117,5 +114,32 @@ void CFRLogger::Log(const cppmicroservices::ServiceReferenceBase& sr,
     currLogger->Log(sr, level, message, ex);
   }
 }
+
+void CFRLogger::Open()
+{
+  auto l = this->Lock();
+  US_UNUSED(l);
+  cfrContext = GetBundleContext();
```

Comment:
however unlikely, `GetBundleContext()` can return an "invalid" bundle context which, if used, will cause failures. This failure mode needs to be handled.